src/Polly.Core/CircuitBreaker/Controller/CircuitStateController.cs

```diff
@@ -308,11 +308,21 @@ private void SetLastHandledOutcome_NeedsLock(Outcome<T> outcome)
     private BrokenCircuitException CreateBrokenCircuitException()
     {
         TimeSpan retryAfter = _blockedUntil - _timeProvider.GetUtcNow();
-        return _breakingException switch
+        var ex = _breakingException switch
         {
             Exception exception => new BrokenCircuitException(BrokenCircuitException.DefaultMessage, retryAfter, exception),
             _ => new BrokenCircuitException(BrokenCircuitException.DefaultMessage, retryAfter)
         };
+
+        _telemetry.UpdateTelemetrySource(ex);
```

Comment:
IMHO this method call is super easy to forget. Also if someone wants to construct any of the Polly exception (for whatever reason) then she has to be aware of this feature and has to use it.